cmd/check_container_test.go

```diff
@@ -21,27 +22,20 @@ import (
 )
 
 var _ = Describe("Check Container Command", func() {
-	var contextTempDir string
-	var artifactsDir string
-
-	BeforeEach(func() {
-		// instantiate err to make sure we can equal-assign in the following line.
-		var err error
-		contextTempDir, err = os.MkdirTemp(os.TempDir(), "check-container-test-*")
-		Expect(err).ToNot(HaveOccurred())
-		Expect(len(contextTempDir)).ToNot(BeZero())
-		artifactsDir = path.Join(contextTempDir, "artifacts")
-		artifacts.SetDir(artifactsDir)
-	})
-
-	AfterEach(func() {
-		err := os.RemoveAll(contextTempDir)
-		Expect(err).ToNot(HaveOccurred())
-		artifacts.Reset()
-	})
+	BeforeEach(artifactsAndLogTemp)
 
 	Context("when running the check container subcommand", func() {
 		Context("With all of the required parameters", func() {
+			BeforeEach(func() {
+				tmpDir, err := os.MkdirTemp("", "cmd-execute-*")
+				Expect(err).ToNot(HaveOccurred())
+				artifacts.SetDir(filepath.Join(tmpDir, "artifacts"))
+				DeferCleanup(os.RemoveAll, tmpDir)
+				os.Setenv("PFLT_ARTIFACTS", artifacts.Path())
+				os.Setenv("PFLT_LOGFILE", filepath.Join(tmpDir, "preflight.log"))
+				DeferCleanup(os.Unsetenv, "PFLT_ARTIFACTS")
+				DeferCleanup(os.Unsetenv, "PFLT_LOGFILE")
```

Comment:
1) Could we have used flags here?
2) Can we organize the Defers together?

Reply:
Possibly... Let me look into that. If not, I'll put the defers together.

Reply:
On the one hand, yes, flags could be used. But then that makes L40 ugly... I think the env route is the correct one. I'll clean up the defers.